lib/src/dialog.js

```diff
@@ -19,9 +19,33 @@
 const electron = require('electron');
 const Bluebird = require('bluebird');
 const zipImage = require('resin-zip-image');
+const homedir = require('user-homedir');
 const packageJSON = require('../../package.json');
 
 /**
+ * @summary Get default directory for file selection dialog.
+ * @function
+ * @private
+ *
+ * @description
+ * This function is meant to be used with electron dialog.showOpenDialog method.
+ *
+ * If etcher is run with sudo, it will return the sudo user's home directory,
+ * as long as the SUDO_USER environment variable is set.
+ *
+ * In every other case the function will return undefined so that electron
+ * decides what the default path should be.
+ *
+ * Works on linux and darwin platforms only.
+ */
+exports.getDefaultSelectionPath = Bluebird.method(function() {
```

Comment:
I believe this function can be removed now that we have all required functionality in `user-homedir`.